library/ssl_msg.c

```diff
@@ -5858,6 +5858,12 @@ static int ssl_handle_hs_message_post_handshake_tls12( mbedtls_ssl_context *ssl
      * - For server-side, expect CLIENT_HELLO.
      * - Fail (TLS) or silently drop record (DTLS) in other cases.
      */
+#if defined(MBEDTLS_SSL_RENEGOTIATION) \
+    || defined(MBEDTLS_SSL_PROTO_TLS1) \
+    || defined(MBEDTLS_SSL_PROTO_TLS1_1) \
+    || defined(MBEDTLS_SSL_PROTO_TLS1_2)
```

Comment:
Those checks seem redundant given that the whole block is guarded by `TLS1_2_OR_EARLIER`.

Reply:
`TLS1_2_OR_EARLIER` include `MBEDTLS_SSL_PROTO_SSL3` . And below usage is not wrapped with `SSL3` . So I did not use `TLS1_2_OR_EARLIER`